hyperspy/io_plugins/tvips.py

```diff
@@ -424,17 +431,18 @@ def file_reader(filename,
     stagebeta = all_metadata[0]["stageb"][0]
     # mag = all_metadata[0]["mag"][0]  # TODO it is unclear what this value is
     focus = all_metadata[0]["objective"][0]
-    sigtype = "diffraction" if mode == 2 else "imaging"
+    # TODO at the moment hyperspy doesn't distinguish between imaging types
+    # sigtype = "diffraction" if mode == 2 else "imaging"
     metadata = {
         'General': {
             'original_filename': os.path.split(filename)[1],
             'date': date,
             'time': time,
             'time_zone': "UTC",
         },
-        "Signal": {
-            'signal_type': sigtype
-        },
+        # "Signal": {
```

Comment:
You can use any signal type listed in https://github.com/hyperspy/hyperspy-extensions-list. In this case, it could be `diffraction` or `electron_diffraction`. If I remember correctly, TVIPS camera can be used for X-rays, so if there is way to make sure that it was an electron diffraction, then it would be more accurate to use `electron_diffraction`. 

For imaging case, it should be left "empty" as there is no extension which implements a corresponding signal.

Reply:
I now only add the `Signal` key if mode is 2 and set `sigtype` to `diffraction`. I don't know whether it's possible to distinguish between electrons or X-rays, but it should be relatively easy for users to change from diffraction to electrondiffraction if relevant.